src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -2495,9 +2495,12 @@ internal void OnManagedConnectionFailed(object sender, ConnectionFailedEventArgs
                     }
                     catch (Exception)
                     {
-
                     }
-                }, null, TimeSpan.FromSeconds(0), TimeSpan.FromSeconds(1));
+                    finally
+                    {
+                        connection.sentinelMasterReconnectTimer?.Change(TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(0));
```

Comment:
If we're disabling period pulling, the second arg should be `Timeout.InfiniteTimeSpan` here :)

Reply:
Yep! -1 is enough